tensorflow/lite/kernels/cpu_backend_context.cc

```diff
@@ -37,7 +44,7 @@ CpuBackendContext* CpuBackendContext::GetFromContext(TfLiteContext* context) {
     // We do the lazy initialization here for the TfLiteInternalBackendContext
     // that's wrapped inside ExternalCpuBackendContext.
     cpu_backend_context = new CpuBackendContext();
-    if (context->recommended_num_threads != -1) {
+    if (IsValidNumThreads(context->recommended_num_threads)) {
       cpu_backend_context->SetMaxNumThreads(context->recommended_num_threads);
```

Comment:
I think the change of this value (i.e. context->recommended_num_threads) is only exposed externally via Interpreter::SetNumThreads where you already added the check. So, it looks to me adding the check here isn't that necessary.

Perhaps we could add a check to see if "recommend_num_threads" is set to -1 that indicates we will set the #(threadpool threads) to a default value (i.e. kDefaultNumThreadpoolThreads you defined here).